src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -625,6 +631,27 @@ internal void Log(GrpcEvent msg)
             }
         }
 
+        internal void LogMetric(GrpcEvent msg)
+        {
+            var rpcMetric = msg.Message.RpcMetric;
+            if (_executingInvocations.TryGetValue(rpcMetric.InvocationId, out ScriptInvocationContext context))
+            {
+                // Restore the execution context from the original invocation. This allows AsyncLocal state to flow to loggers.
+                System.Threading.ExecutionContext.Run(context.AsyncExecutionContext, (s) =>
+                {
+                    var metricDetails = JsonSerializer.Deserialize<IDictionary<string, object>>(rpcMetric.Properties);
+                    if (metricDetails.TryGetValue<JsonElement>(LogConstants.NameKey, out var metricNameElt)
```

Comment:
We shouldn't pull Name/Value out of the properties -- `RpcMetric` has those directly on it. `Properties` should be for anything extra that you may want to pass along (such as some of the special keys that App Insights knows about).